tests/unit/pipeline/test_ner_re_pipeline.py

```diff
@@ -0,0 +1,390 @@
+import dataclasses
+
+import pytest
+from pie_modules.annotations import BinaryRelation, LabeledSpan
+from pie_modules.documents import TextBasedDocument
+from pytorch_ie import AnnotationLayer
+from pytorch_ie.core import annotation_field
+from pytorch_ie.models import (
+    TransformerSpanClassificationModel,
+    TransformerTextClassificationModel,
+)
+from pytorch_ie.taskmodules import (
+    TransformerRETextClassificationTaskModule,
+    TransformerSpanClassificationTaskModule,
+)
+
+from src.pipeline.ner_re_pipeline import (
+    NerRePipeline,
+    add_annotations_from_other_documents,
+    clear_annotation_layers,
+    move_annotations_from_predictions,
+    move_annotations_to_predictions,
+    process_documents,
+    process_pipeline_steps,
+)
+
+
+@dataclasses.dataclass
+class TextDocumentWithLabeledSpansAndBinaryRelations(TextBasedDocument):
+    labeled_spans: AnnotationLayer[LabeledSpan] = annotation_field(target="text")
+    binary_relations: AnnotationLayer[BinaryRelation] = annotation_field(target="labeled_spans")
+
+
+@pytest.fixture
+def document():
+    document = TextDocumentWithLabeledSpansAndBinaryRelations(
+        text="Harry lives in Berlin, Germany. He works at DFKI.", id="tmp"
+    )
+    document.labeled_spans.predictions.extend(
+        [
+            LabeledSpan(start=0, end=5, label="PERSON"),
+            LabeledSpan(start=44, end=48, label="ORGANIZATION"),
+        ]
+    )
+
+    assert str(document.labeled_spans.predictions[0]) == "Harry"
+    assert str(document.labeled_spans.predictions[1]) == "DFKI"
+
+    document.labeled_spans.extend(
+        [
+            LabeledSpan(start=0, end=5, label="PERSON"),
+            LabeledSpan(start=15, end=30, label="LOCATION"),
+            LabeledSpan(start=44, end=48, label="ORGANIZATION"),
+        ]
+    )
+    assert str(document.labeled_spans[0]) == "Harry"
+    assert str(document.labeled_spans[1]) == "Berlin, Germany"
+    assert str(document.labeled_spans[2]) == "DFKI"
+
+    return document
+
+
+@pytest.fixture
+def document_with_relations(document):
+
+    document.binary_relations.predictions.extend(
```

Comment:
is it really ok to directly add the relations to the document instead of creating a copy and modifying that? I would guess, the current code modifies also the `document` fixture? But I haven't double checked that, just my gut feeling.

Reply:
You're right, it does modifies the document fixture. So if I understood it right, your suggestion is to copy the document before modifying ?

Reply:
yes. just add `document = document.copy()` to the beginning of this fixture.